browser/src/slideshow/CanvasLoader.ts

```diff
@@ -11,12 +11,25 @@
  */
 
 /*
- * Loader for slideshow
+ * Loader for slideshow - it shows animation for the user to indicate loading
  */
 
 declare var SlideShow: any;
 
-class CanvasLoader extends Transition2d {
+abstract class CanvasLoader {
+	public abstract startLoader(): void;
+	public abstract stopLoader(): void;
+}
+
+class CanvasLoader2d implements CanvasLoader {
+	constructor(transitionParameters: TransitionParameters) {} // eslint-disable-line
+
+	public startLoader(): void {} // eslint-disable-line
+
+	public stopLoader(): void {} // eslint-disable-line
+}
+
+class CanvasLoaderGl extends Transition2d implements CanvasLoader {
 	private animationId: number | null = null;
 
 	constructor(transitionParameters: TransitionParameters) {
```

Comment:
What are transitionParameters for ? Above  CanvasLoader base class has neither constructor nor properties.

Reply:
Oh, I've missed `extends Transition2d` for the GL version; sorry for the noise.